tests/unit/test_cmr_search.py

```diff
@@ -238,4 +249,87 @@ def test_exceptions_granule_link(self):
         with self.subTest('Granule has `links` and all correct attributes'):
             self.assertEqual(get_granule_link(granule_response_links_correct),
                              'https://data.gesdisc.earthdata.nasa.gov/.nc4')
-    
\ No newline at end of file
+    
+    def test_granule_name(self):
+        ''' Check if get_granule_name returns expected string for response
+            with key "producer_granule_id" and raises IndexError for response
+            without a "producer_granule_id"
+        '''
+        granule_response = [{
+            'producer_granule_id': 'MERRA2_400.tavg1_2d_slv_Nx.20210601.nc4',
+            'boxes': ['-90 -180 90 180'],
+            'time_start': '2021-06-01T00:00:00.000Z'}]
+        
+        granule_response_no_producer_id = [{
+            'boxes': ['-90 -180 90 180'],
+            'time_start': '2021-06-01T00:00:00.000Z'}]
+        
+        with self.subTest('Granule has "producer_granule_id"'):
+            self.assertEqual(get_granule_name(granule_response),
+                             'MERRA2_400.tavg1_2d_slv_Nx.20210601.nc4')
+        
+        with self.subTest('Granule does not have "producer_granule_id"'):
+            with self.assertRaises(IndexError):
+                get_granule_name(granule_response_no_producer_id)
+    
+    
+    def test_download_gran_exceptions(self):
+        ''' Check if exceptions are raised for incorrect input parameters
+        '''
+            
+        with self.assertRaises(ValueError):
+            # Input parameter, `granule_link` should be string
+            download_granule(5, 'foo')
+            
+        with self.assertRaises(ValueError):
+            # Input parameter, `out_filename` should be string
+            download_granule('foo', 10)
+        
+        with self.assertRaises(ValueError):
+            # Input parameter, `out_filename` should be string
+            download_granule('foo', 10)
+        
+        with self.assertRaises(RequestsException):
+            # Not a real https link
+            download_granule('foo', 'foo.nc4')
+    
+    
+    def _mock_requests(self, status=200, content="CONTENT"):
```

Comment:
Okay, I am pretty proud of this one. I wanted to mock `requests.get` and thought it would be easier to write a function that returns this mock object, instead of creating the mocked object in the same test for checking the mocked object against download_granules. Also, I noticed in some of your harmony code you start functions that are only used once with a `_` ?? I was reading somewhere that this is convention but feel free to correct me if I am wrong.